plugins/modules/mso_backup.py

```diff
@@ -0,0 +1,167 @@
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: cisco.mso.modules
+'''
+
+EXAMPLES = r'''
+- name: Create a new backup
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    backup: Backup
+    description: via Ansible
+    location_type: local
+    state: present
+  delegate_to: localhost
+- name: Remove a Backup
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    backup: Backup
+    state: absent
+  delegate_to: localhost
+- name: Query a backup
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    backup: Backup
+    state: query
+  delegate_to: localhost
+  register: query_result
+- name: Query all backups
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, issubset
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        location_type=dict(type='str'),
+        description=dict(type='str'),
+        backup=dict(type='str', aliases=['name']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    description = module.params.get('description')
+    location_type = module.params.get('location_type')
+    state = module.params.get('state')
+    backup = module.params.get('backup')
+
+    mso = MSOModule(module)
+
+    backup_id = None
+    path = 'backups'
+    count_backups = 0
+    mso.existing = mso.query_objs('backups')
+    if backup:
+        if mso.existing:
+            data = mso.existing
+            mso.existing = []
+            for index_of_name in range(0, len(data)):
+                backup_id = data[index_of_name]['id']
+                if (str(backup) + '_' + backup_id == data[index_of_name]['metadata']['name']):
+                    path = 'backups/{id}'.format(id=backup_id)
+                    count_backups += 1
+                    mso.existing.append(data[index_of_name])
+            if count_backups < 1:
+                mso.existing = mso.get_obj(path, name=backup)
+
+    if state == 'query':
+        pass
+        mso.exit_json()
+
+    if state == 'absent':
+        mso.previous = mso.existing
+        if mso.existing:
+            if module.check_mode:
+                mso.existing = {}
+            else:
+                mso.existing = mso.request(path, method='DELETE')
+        mso.exit_json()
+
+    path = 'backups'
```

Comment:
This looks like line 115 and 117 why redo this here? Why also not using get_obj before and just using query_objs?

Reply:
I think mso.get_obj(path, name=backup) is empty because the name attribute in the response body is in metadata and not with id of the backup. Hence, I use query_objs..

"backups": [
    {
      "id": "20200718030540",
      "sizeKB": 790.19,
      "timestamp": "2020-07-18T03:05:40",
      "metadata": {
        "description": "",
        "name": "Backup100_20200718030540"
      }
    }

Reply:
path = 'backups' is like a reset button in this case. We need this because we only use Post method, which requires the main path. If we had Put/Patch we could've avoided it

Reply:
But then why do you use it here. It will clear out the mso.existing which you already have?
If path is always backup for the POST, just put it in the line 161 mso.request you don't need a variable single use.